src/hotspot/share/c1/c1_GraphBuilder.cpp

```diff
@@ -2537,7 +2537,11 @@ void GraphBuilder::new_instance(int klass_index) {
   bool will_link;
   ciKlass* klass = stream()->get_klass(will_link);
   assert(klass->is_instance_klass(), "must be an instance klass");
-  NewInstance* new_instance = new NewInstance(klass->as_instance_klass(), state_before, stream()->is_unresolved_klass());
+  ciKlass* ik = klass;
+  if (ik == compilation()->env()->Object_klass() && compilation()->env()->Object_klass()->is_abstract()) {
```

Comment:
Does adding if EnableValhalla as the first condition, or a "// Valhalla" comment, make this clearer ? There are a few places in the change set with the same review comment, but I won't repeat.